controllers/workspace/provision/automount_test.go

```diff
@@ -0,0 +1,190 @@
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package provision
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/devfile/devworkspace-operator/apis/controller/v1alpha1"
+)
+
+type mountedVolumeType int
+
+const (
+	devWorkspaceVolume mountedVolumeType = iota
+	secretVolumeType
+	configMapVolumeType
+)
+
+const testContainerName = "testContainer"
+
+func TestCheckAutoMountVolumesForCollision(t *testing.T) {
+	type volumeDesc struct {
+		name       string
+		mountPath  string
+		volumeType mountedVolumeType
+	}
+	tests := []struct {
+		name                  string
+		basePodAdditions      []volumeDesc
+		automountPodAdditions []volumeDesc
+		errRegexp             string
+	}{
+		{
+			name: "Does not error when mounts are valid",
+			basePodAdditions: []volumeDesc{
+				{
+					name:       "baseVolume",
+					mountPath:  "basePath",
+					volumeType: configMapVolumeType,
+				},
+			},
+			automountPodAdditions: []volumeDesc{
+				{
+					name:       "automountConfigMap",
+					mountPath:  "/configmap/mount",
+					volumeType: configMapVolumeType,
+				},
+				{
+					name:       "automountSecret",
+					mountPath:  "/secret/mount",
+					volumeType: secretVolumeType,
+				},
+			},
+		},
+		{
+			name: "Detects volume name collision",
+			basePodAdditions: []volumeDesc{
+				{
+					name:       "baseVolume",
+					mountPath:  "basePath",
+					volumeType: configMapVolumeType,
+				},
+			},
+			automountPodAdditions: []volumeDesc{
+				{
+					name:       "baseVolume",
+					mountPath:  "/configmap/mount",
+					volumeType: configMapVolumeType,
+				},
+			},
+			errRegexp: "DevWorkspace volume baseVolume conflicts with automounted volume from configmap baseVolume",
+		},
+		{
+			name: "Detects mountPath collision with DevWorkspace",
+			basePodAdditions: []volumeDesc{
+				{
+					name:       "baseVolume",
+					mountPath:  "/collision/path",
+					volumeType: devWorkspaceVolume,
+				},
+			},
+			automountPodAdditions: []volumeDesc{
+				{
+					name:       "testVolume",
+					mountPath:  "/collision/path",
+					volumeType: secretVolumeType,
+				},
+			},
+			errRegexp: fmt.Sprintf("DevWorkspace volume baseVolume in container %s has same mountpath as auto-mounted volume from secret testVolume", testContainerName),
+		},
+		{
+			name: "Detects mountPath collision in automounted volumes",
+			automountPodAdditions: []volumeDesc{
+				{
+					name:       "testVolume1",
+					mountPath:  "/test/mount",
+					volumeType: secretVolumeType,
+				},
+				{
+					name:       "testVolume2",
+					mountPath:  "/test/mount",
+					volumeType: configMapVolumeType,
+				},
+			},
+			errRegexp: "auto-mounted volumes from configmap testVolume2 and secret testVolume1 have the same mount path",
```

Comment:
Maybe it makes sense to take in quotes objects names?

Reply:
Done :)